lib/stdlib/src/timer.erl

```diff
@@ -601,12 +601,12 @@ do_apply({erlang, exit, [Name, Reason]}, _) ->
 do_apply({M,F,A}, false) ->
     try spawn(M, F, A)
     of _ -> {ok, spawn}
-    catch _:_ -> error
+    catch error:badarg -> error
```

Comment:
I just noticed... `spawn` only fails with `badarg` if the arguments are _way_ off, ie if `M` or `F` are not atoms or if `A` is not a list. Pretty much everything else, like the length of `A` not matching the arity of `M:F`, will not cause any exception thrown by the `spawn` itself, but instead a process _will_ be started and the exception will be raised from there.

Given that it is impossible (famous last words :sweat_smile:) to start `apply_*` timers where `M` and `F` are _not_ atoms or where `A` is _not_ a list ***when going through the `timer` API functions***, ie without (deliberately) ***bypassing*** them by sending messages to the timer server directly, maybe the `spawn`s should not be wrapped in `try ... catch`es at all?

Reply:
Yes, the try/catch is probably unnecessary. Since the PR targeted a patch release, I kept the try/catch just in case to minimize the risk of introducing a bug.